pkg/asset/machines/azure/machines.go

```diff
@@ -361,20 +363,41 @@ func ConfigMasters(machines []machineapi.Machine, controlPlane *machinev1.Contro
 	return nil
 }
 
-func getNetworkInfo(platform *azure.Platform, clusterID, role string) (string, string, string, error) {
+func getNetworkInfo(platform *azure.Platform, clusterID, role string, subnetZones int) (string, string, []string, error) {
 	networkResourceGroupName := platform.NetworkResourceGroupName
 	if platform.VirtualNetwork == "" {
 		networkResourceGroupName = platform.ClusterResourceGroupName(clusterID)
 	}
-
+	virtualNetworkName := platform.VirtualNetworkName(clusterID)
+	var subnetRole capz.SubnetRole
+	var defaultSubnet string
 	switch role {
 	case "worker":
-		return networkResourceGroupName, platform.VirtualNetworkName(clusterID), platform.ComputeSubnetName(clusterID), nil
-	case "master":
-		return networkResourceGroupName, platform.VirtualNetworkName(clusterID), platform.ControlPlaneSubnetName(clusterID), nil
+		subnetRole = capz.SubnetNode
+		defaultSubnet = platform.ComputeSubnetName(clusterID)
+	case controlPlaneRoleName:
+		subnetRole = capz.SubnetControlPlane
+		defaultSubnet = platform.ControlPlaneSubnetName(clusterID)
 	default:
-		return "", "", "", fmt.Errorf("unrecognized machine role %s", role)
+		return "", "", nil, fmt.Errorf("unrecognized machine role %s", role)
+	}
+
+	subnets := []string{}
+	for _, subnetSpec := range platform.Subnets {
+		if subnetSpec.Role == subnetRole {
+			subnets = append(subnets, subnetSpec.Name)
+		}
+	}
+
+	if len(subnets) == 0 {
+		subnets = append(subnets, defaultSubnet)
```

Comment:
Unclear what we are tying to accomplish here. If the role is control plane, subnets will have 1 defaultSubnet of the name `$infraid-master-subnet`. For worker subnets, there would be 1 subnet with name `$infraid-worker-subnet` and then `$infraid-worker-subnet-2` and so on. Is that what we want?

Reply:
In a default install with none of this, there should be a default control plane and compute subnet. For multi zone, there should be x compute subnets for each availability zone. All that for loop stuff later is just for naming purposes, I wanted the names to be like subnet, subnet-2, subnet-3 ... instead of subnet-1,subnet-2, subnet-3. Its just for readability.